torch/csrc/jit/runtime/static/impl.cpp

```diff
@@ -104,110 +104,36 @@ StaticRuntime::StaticRuntime(const torch::jit::Module& m)
     }
   }
 
-  // fill constant_table_ and operator_table_
+  // fill workspace_ with constants
   for (Node* node : graph_->nodes()) {
-    switch (node->kind()) {
-      case prim::Constant:
-        CHECK(node->output()->type()->kind() != FunctionType::Kind);
-        constant_table_[node->output()] = toIValue(node->output()).value();
-        break;
-      case prim::ListConstruct:
-        nodes_.emplace_back(node, nullptr);
-        break;
-      case prim::TupleConstruct:
-        nodes_.emplace_back(node, nullptr);
-        break;
-      default: {
-        const Operator& op = node->getOperator();
-        CHECK(op.hasOperation());
-        nodes_.emplace_back(node, op.getOperation(node));
-      }
-    }
-  }
-}
-
-void StaticRuntime::getInputIValues(
-    Node* node,
-    const ConstantMap& ws,
-    std::vector<IValue>& stack) const {
-  const size_t size = node->inputs().size();
-  stack.reserve(size);
-  for (size_t i = 0; i < size; i++) {
-    Value* v = node->inputs()[i];
-    auto f = constant_table_.find(v);
-    if (f == constant_table_.end()) {
-      auto f_ws = ws.find(v);
-      TORCH_CHECK(
-          f_ws != ws.end(),
-          "Workspace does not contain Value ",
-          v->debugName());
-      stack.emplace_back(f_ws->second);
+    if (node->kind() == prim::Constant) {
+      CHECK(node->output()->type()->kind() != FunctionType::Kind);
+      workspace_[node->output()] = toIValue(node->output()).value();
```

Comment:
It's important to keep the constants(weights) separate from activations because weights will be accessed by all threads and activations are per thread.

Reply:
let's handle that separately -- perhaps by mutating the `value()` of `prim::Constant` nodes across model copies in a multithreaded domain.

Reply:
It was handled pretty cleanly in the old version. Why can't we just keep them in separate tables?

Reply:
actually, now that I think about it, I thought we would have a single StaticRuntime per thread -- isn't this why we removed the deep copy?

I'm not entirely sure I understand why these need to be two separate tables.  They seem almost identical to me

Reply:
note that Tensors can be thread local but the data backing them can exist separately -- i.e. each workspace of each thread can have the same constants without copying weights around

Reply:
No we should not have a single StaticRuntime per thread. The number of threads launched at model runtime is dynamic, depending on the total number of ads to rank. Also, you want to do model loading and model optimization only once, not for every thread.

The point of having weights separate from activations is that weights are readonly and accessible to all threads. Activations are threadlocal and mutable. I think having a clear separation is a lot easier and cleaner.

Reply:
it'd be super useful if we had a mockup of this interaction model -- how are threads launched? do we have to provide an interface for that or just use whatever threads are available somehow?

Reply:
I'll add a test when I get time. Let's come back to this topic once I have that.